encoding/ssz/query/generalized_index.go

```diff
@@ -0,0 +1,274 @@
+package query
+
+import (
+	"errors"
+	"fmt"
+)
+
+const (
+	bytesPerChunk = 32
+	bitsPerChunk  = 256
+	listBaseIndex = 2
+)
+
+// GetGeneralizedIndexFromPath calculates the generalized index for a given path.
+// To calculate the generalized index, two inputs are needed:
+// 1. The sszInfo of the root info, to be able to navigate the SSZ structure
+// 2. The path to the field (e.g., "field_a.field_b[3].field_c")
+// It walks the path step by step, updating the generalized index at each step.
+func GetGeneralizedIndexFromPath(info *SszInfo, path []PathElement) (uint64, error) {
+	if info == nil {
+		return 0, errors.New("sszInfo is nil")
+	}
+
+	// If path element list is empty, no generalized index can be computed.
+	if len(path) == 0 {
+		return 0, errors.New("cannot compute generalized index for an empty path")
+	}
+
+	// Starting from the root generalized index
+	root := uint64(1)
+	currentInfo := info
+
+	for _, pathElement := range path {
+		element := pathElement
+
+		// If we descend to a basic type, the path cannot continue further
+		if isBasicType(currentInfo.sszType) {
+			return 0, fmt.Errorf("cannot descend into basic type %s for path element %q", currentInfo.sszType, pathElement.Name)
+		}
+
+		// Check that we are in a container to access fields
+		if currentInfo.sszType != Container {
```

Comment:
Isn't having only this check sufficient? Do we need the above `isBasicType` check?